app/models/invite_code.rb

```diff
@@ -0,0 +1,23 @@
+require 'coupon_code'
+
+class InviteCode < CouchRest::Model::Base
+  use_database 'invite_codes'
+  property :invite_code, String, :read_only => true
+  property :invite_count, Integer, :default => 0, :accessible => true
+
+  timestamps!
+
+  design do
+    view :by__id
+    view :by_invite_code
+    view :by_invite_count
+  end
+
+  def initialize(attributes = {}, options = {})
+    super(attributes, options)
+    write_attribute('invite_code', CouponCode.generate) if new?
```

Comment:
cool. read_only and then use write_attribute. Took me a second to figure out why not `self.invite_code =` - seems like a good idea.